src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -210,29 +203,52 @@ protected Object fetchField(ExecutionContext executionContext, ExecutionStrategy
 
         InstrumentationFieldFetchParameters instrumentationFieldFetchParams = new InstrumentationFieldFetchParameters(executionContext, fieldDef, environment);
         InstrumentationContext<Object> fetchCtx = instrumentation.beginFieldFetch(instrumentationFieldFetchParams);
-        Object fetchedValue = null;
+        CompletableFuture<?> fetchedValue;
+        DataFetcher dataFetcher = fieldDef.getDataFetcher();
+        dataFetcher = instrumentation.instrumentDataFetcher(dataFetcher, instrumentationFieldFetchParams);
         try {
-            DataFetcher dataFetcher = fieldDef.getDataFetcher();
-            dataFetcher = instrumentation.instrumentDataFetcher(dataFetcher, instrumentationFieldFetchParams);
-            fetchedValue = dataFetcher.get(environment);
-
-            fetchCtx.onEnd(fetchedValue);
+            Object fetchedValueRaw = dataFetcher.get(environment);
+            if (fetchedValueRaw instanceof CompletableFuture) {
```

Comment:
I think this should accept a CompletionStage at this specific point.

While the JDK only has 1 implementation, people can write their own.

http://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletionStage.html#toCompletableFuture--

The toCompletableFuture is provided as a interop to allow 2 different CompletionStages to interop.

This would be a tiny minor thing to be honest since the work around is for them to call toCompletableFuture() themselves in the DF.